src/cluster/slot_migrate.cc

```diff
@@ -1110,3 +1162,224 @@ void SlotMigrator::resumeSyncCtx(const Status &migrate_result) {
     blocking_context_ = nullptr;
   }
 }
+
+void SlotMigrator::SetMigrateBatchRateLimit(size_t bytes_per_sec) {
+  migrate_batch_bytes_per_sec_ =
+      (bytes_per_sec == 0 || bytes_per_sec > kMaxMigrateBatchRate) ? kMaxMigrateBatchRate : bytes_per_sec;
+  migrate_batch_rate_limiter_->SetBytesPerSecond(static_cast<int64_t>(migrate_batch_bytes_per_sec_));
+}
+
+MigrateType SlotMigrator::GetMigrateType() { return static_cast<MigrateType>(svr_->GetConfig()->migrate_type); }
+
+Status SlotMigrator::getClockSkew(int64_t *diff_us) {
+  assert(dst_redis_context_ != nullptr);
+  uint64_t send_timestamp = util::GetTimeStampUS();
+  auto *reply = static_cast<redisReply *>(redisCommand(dst_redis_context_, "TIME"));
+  uint64_t receive_timestamp = util::GetTimeStampUS();
+
+  auto exit = MakeScopeExit([reply] { freeReplyObject(reply); });
+
+  if (dst_redis_context_->err != 0) {
+    return {Status::NotOK, std::string(dst_redis_context_->errstr)};
+  }
+
+  if (reply == nullptr) {
+    return {Status::NotOK, "get null reply from TIME command"};
+  }
+
+  if (reply->type == REDIS_REPLY_ERROR) {
+    auto error_str = std::string(reply->str);
+    return {Status::NotOK, error_str};
+  }
+
+  if (reply->elements != 2) {
+    return {Status::NotOK, "get invalid reply from TIME command"};
+  }
+
+  uint64_t dst_timestamp = std::stoul(reply->element[0]->str) * 1000000 + std::stoul(reply->element[1]->str);
```

Comment:
stoul will throw exceptions while the parsing is failed, so i wonder if the parsing will always succeed here?

Reply:
Yes, normal `TIME` command will parse successfully.